Sources/Query.swift

```diff
@@ -8,159 +8,507 @@
+/// so that all concrete types, `Query`, `AssetQuery`, `FilterQuery`, and `QueryOn<EntryType>`, can use the same implementation.
+public protocol ChainableQuery: AbstractQuery {}
+public extension ChainableQuery {
+
+    /**
+     Convenience initializer for a select operation query in which only the fields specified in the fieldNames property will be returned in the JSON response.
+     The `"sys"` dictionary is always requested by the SDK. 
+     Note that if you are using the select operator with an instance `QueryOn<EntryType>`
+     that you must make properties that you are ommitting in the response (by not including them in your selections array) optional properties.
+     Example usage:
+     
+     ```
+     let query = try! QueryOn<Cat>(selectingFieldsNamed: ["fields.bestFriend", "fields.color", "fields.name"])
+     client.fetchEntries(with: query).observable.then { cats in
+        // Do stuff with cats.
+     }
+     ```
+
+     See: <https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/select-operator>
+     - Parameter fieldNames: An array of field names to include in the JSON response.
+     - Parameter locale: An optional locale argument to return localized results. If unspecified, the locale originally
+     set on the `Client` instance is used.
+     - Throws: Will throw an error if property names are not prefixed with `"fields."`, if selections go more than 2 levels deep
+               ("fields.bestFriend.sys" is not valid), or if more than 99 properties are selected.
+    */
+    public init(selectingFieldsNamed fieldNames: [String], for locale: String? = nil) throws {
+        self.init()
+        try self.select(fieldsNamed: fieldNames, locale: locale)
+    }
 
-    /// Query operation
-    public static func query(where name: String, _ operation: QueryOperation, for locale: String? = nil) -> Query<ContentType> {
-        let query = Query<ContentType>()
-        return query.query(where: name, operation, for: locale)
+    /**
+     Convenience initializer for a ordering responses by the values at the specified field. Field types that can be
+     specified are Strings, Numbers, or Booleans.
+
+     Example usage:
+
+     ```
+     let query = try! Query(orderedBy: "sys.createdAt")
+
+     client.fetchEntries(with: query).observable.then { entries in
+        // Do stuff with entries.
+     }
+     ```
+
+     See: <https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/order>
+     and: <https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/order-with-multiple-parameters>
+     - Parameter propertyName: One or more properties on the Resource by which the results will be ordered.
+     - Parameter reverse: An Bool specifying if the returned order should be reversed or not. Defaults to `false`.
```

Comment:
I am confused what would happen if I specify `propertyName: ['cat', 'dog']` and `reverse:true` would the first one be reversed, the second or all?